### ISSUE TITLE
[Filter Panel]: Deprecate <dds-filter-panel-heading> component

### ISSUE THREAD

**Issue by m4olivei** [carbon-design-system/carbon-for-ibm-dotcom#11062]
### The problem

The `<dds-filter-panel-heading>` component is presently required to be used with `<dds-filter-panel-composite>`, and is effectively hidden, its child nodes (typically a text node) harvested via the `slotchange` event, for use composed with the number of selected values.

Slots are best suited for things that are directly reflected into the shadow DOM. It's current use results in an over complicated, hard to follow implementation.

### The solution

Introduce a new `heading` attribute for the `<dds-filter-panel-composite>` component to fulfill the same purposes as the `<dds-filter-panel-heading>` component. Mark the `<dds-filter-panel-heading>` component as deprecated, to be removed in the next major version.

### Application/website

ibm.com

### Business priority

Low Priority = release date is not dependent on fix or not upcoming

### What time frame would this ideally be needed by (if applicable)

No hard dates here.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

**Comment by m4olivei** [comment]
> @[kennylam](https://github.com/kennylam) kennylam closed this as [completed](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues?q=is%3Aissue+is%3Aclosed+archived%3Afalse+reason%3Acompleted) [8 minutes ago](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/11062#event-11628663352)

@kennylam I think this is still an open issue? Unless we're closing as "won't fix"?

**Comment by kennylam** [comment]
Oh sorry about that. I thought it was fixed in the referenced PR.

**Comment by m4olivei** [comment]
No worries. Thanks. I'll come back around to this on a rainy day.
